app/Domain/Clients/Controllers/ShowClient.php

```diff
@@ -52,38 +54,95 @@ public function init(
             }
         }
 
+
+        public function get($params): Response
+        {
+            Auth::authOrRedirect([Roles::$owner, Roles::$admin, Roles::$manager], true);
+
+            $id = '';
+
+            if (isset($params['id']) === true) {
+                $id = (int) ($params['id']);
+            }
+
+            $row = $this->clientService->get($id);
+
+            if ($row === false) {
+                $this->tpl->display('errors.error404');
+
+                return $this->tpl->display('errors.error403');
+            }
+
+            $clientValues = (object) [
+                'id' => $row->id,
+                'name' => $row->name,
+                'street' => $row->street,
+                'zip' => $row->zip,
+                'city' => $row->city,
+                'state' => $row->state,
+                'country' => $row->country,
+                'phone' => $row->phone,
+                'internet' => $row->internet,
+                'email' => $row->email,
+            ];
+
+
+            if (empty($row) === false && Auth::userIsAtLeast(Roles::$admin)) {
+                $file = app()->make(FileRepository::class);
+                $project = app()->make(ProjectRepository::class);
```

Comment:
We are already passing in the projectService through the constructor, can you use that instead of the repo?